### PR TITLE
implement python version of build_model method for seq2seq

### PR DESCRIPTION
**Pull request by dding3**
Fix https://github.com/intel-analytics/analytics-zoo/issues/1124

### CODE REVIEW

**Comment by dding3** [comment]
Jekins passed. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1817/

**Comment by jason-dai** [comment]
LGTM

File: pyzoo/zoo/pipeline/api/keras/layers/torch.py [hunk 1]
```diff
@@ -788,3 +788,21 @@ def __init__(self, output_height, output_width, align_corner=False,
...
                                              **kwargs)
+
+
+class SelectTable(ZooKerasLayer):
+    """
+    Creates a module that takes a list of JTensors as input and outputs the element at index `index`
+
+    # Arguments
+    index: the index to be selected. 0-based index
+    input_shape: A shape tuple, not including batch.
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras/layers/torch.py]: should be a list of shape tuples?

File: pyzoo/zoo/models/seq2seq/seq2seq.py [hunk 2]
```diff
@@ -172,18 +173,46 @@ class Seq2seq(ZooModel):
...
-        super(Seq2seq, self).__init__(None, bigdl_type,
-                                      encoder,
-                                      decoder,
-                                      list(input_shape) if input_shape else None,
-                                      list(output_shape) if output_shape else None,
-                                      bridge,
-                                      generator)
+        self.encoder = encoder
+        self.decoder = decoder
+        self.input_shape = list(input_shape)
```
Comment (hkvision) [inline pyzoo/zoo/models/seq2seq/seq2seq.py]: originally in the constructor you wrote `list(input_shape) if input_shape else None`, wondering whether the if branch is still needed here.

**Comment by hkvision** [comment]
LGTM

File: pyzoo/zoo/models/seq2seq/seq2seq.py (hunk 2 above)
Comment (dding3) [reply pyzoo/zoo/models/seq2seq/seq2seq.py]: As we don't support input_shape as None, add a little enhancement here(added require input_shape != None logic).

**Comment by dding3** [comment]
Jekins passed. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1824/

Thanks for your review.

**Pull request merged**